configuration/redis/metadata.yaml

```diff
@@ -20,27 +20,93 @@ metadata:
     description: Password for the Redis host
     example:  "KeFg23!"
     type: string
+  - name: redisUsername
+    required: false
+    description: Username for Redis host. Defaults to empty. Make sure your redis server version is 6 or above, and have created acl rule correctly.
+    example: "default"
+    type: string
+  - name: redisType
+    required: false
+    description: |
+      The type of redis. There are two valid values, one is "node" for single node mode, the other is "cluster" for redis cluster mode. Defaults to "node".
+    example: "cluster"
+    type: string
   - name: enableTLS
     required: false
     description: | 
       If enabled, connects to Redis using TLS with public certificates.
     example: "false"
     default: "false"
     type: bool
-  - name: maxRetries
+  - name: redisMaxRetries
     required: false
     description: |
       Maximum number of retries before giving up. Defaults to "3".
     default: "3"
     example: "5"
     type: number
-  - name: maxRetryBackoff
+  - name: redisMinRetryInterval
+    required: false
+    description: |
+      Minimum backoff for redis commands between each retry. Default is "8ms"; "-1" disables backoff.
+    example: "8ms"
+    type: duration
+  - name: redisMaxRetryInterval
+    required: false
+    description: |
+      Maximum backoff for redis commands between each retry. Default is "2s";"-1" disables backoff.
+    example: "5s"
+    type: duration
```

Comment:
I think it is better to explicitly add a `default` field to these methods and remove the default from the description. It would add more value to automated tools and for auto-doc generation.

I just did some of these fields on https://github.com/dapr/components-contrib/pull/2796. Might be convenient to just copy and paste from there to keep consistency. Do notice that not every field used by bindings is required for configuration so there might be a few extra fields you might want to remove if you do copy from there.

I would copy some of these fields from there.